internal/xds/clients/xdsclient/ads_stream.go

```diff
@@ -707,69 +702,79 @@ func resourceNames(m map[string]*xdsresource.ResourceWatchState) []string {
 	return ret
 }
 
-// adsFlowControl implements ADS stream level flow control that enables the
-// transport to block the reading of the next message off of the stream until
-// the previous update is consumed by all watchers.
+// adsFlowControl implements ADS stream level flow control that enables the ADS
+// stream to block the reading of the next message until the previous update is
+// consumed by all watchers.
 //
-// The lifetime of the flow control is tied to the lifetime of the stream.
+// The lifetime of the flow control is tied to the lifetime of the stream. When
+// the stream is closed, it is the responsibility of the caller to set the
+// pending state to false. This ensures that any goroutine blocked on the flow
+// control's wait method is unblocked.
```

Comment:
> When the stream is closed, it is the responsibility of the caller to set the pending state to false. This ensures that any goroutine blocked on the flow control's wait method is unblocked.

I have a question about this. My understanding is that calling `adsFlowControl.stop()` unblocks any goroutines waiting on `adsFlowControl.wait()`. Why is it also the caller's responsibility to set the pending state to `false`? It seems like `stop()` would already handle this.

Reply:
Yeah, I was changing back and forth (experimenting with a few fixes). I guess I failed to update this docstring. Thanks for catching it.